### PR TITLE
[smartswitch] Add HLD for IP address assignment in Smart Switch.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**

### CODE REVIEW

**Review by Yakiv-Huryk — APPROVED** [review]

**Comment by r12f** [comment]
hi Oleksandr, do you mind to move the files from dash folder to "smart-switch" folder? the dash folder is mostly only focus on the dash object model that used by our northbound interface. and everything else related to smart switch will be better to put under smart-switch folder, e.g., HA design.

File: doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+#### Smart Switch configuration ####
+
+Based on the preset `t1-smartswitch` default topology the configuration generated by the sonic-cfggen utility will be the following:
+
+```json
+{
+    "DEVICE_METADATA": {
+        "localhost": {
+            "switch_type": "switch",
+            "type": "SmartSwitch",
```
Comment (prsunny) [inline doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md:148]: This must be subtype. type is used as "LeafRouter" etc.

File: doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        }
+    },
+    "MID_PLANE_BRIDGE" {
+        "GLOBAL": {
+            "bridge" : "bridge_midplane",
+            "address": "169.254.200.254/24",
+            "members": ["Dpu0", "Dpu1"]
```
Comment (prsunny) [inline doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md:155]: Is this really required?

File: doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+## Testing Requirements/Design ##
+
+### Unit Test cases ###
+
+1. Add new YANG model tests for the new MID_PLANE_BRIDGE and DPUS table.
+2. Extend existing YANG model tests to cover MGMT_INTERFACE table changes.
+3. Add new tests to cover the handling of MID_PLANE_BRIDGE and DPUS tables in sonic-cfggen utility.
+4. Add new tests to cover DHCP server configuration generation for the Smart Switch in sonic-cfggen utility.
+5. Extend interface-config.service tests to verify midplane bridge configuration generation.
```
Comment (prsunny) [inline doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md:364]: I think we should have a test to ensure these midplane IPs are not advertised via BGP

**Comment by oleksandrivantsiv** [comment]
> hi Oleksandr, do you mind to move the files from dash folder to "smart-switch" folder? the dash folder is mostly only focus on the dash object model that used by our northbound interface. and everything else related to smart switch will be better to put under smart-switch folder, e.g., HA design.

Done

File: doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md (hunk 1 above)
Comment (oleksandrivantsiv) [reply doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md]: done

**Comment by oleksandrivantsiv** [comment]
@prsunny can we merge this PR?

**Review by prsunny — APPROVED** [review]

**Pull request merged**